rows.go

```diff
@@ -400,116 +410,98 @@ func downloadChunkHelper(ctx context.Context, scd *snowflakeChunkDownloader, idx
 
 	resp, err := scd.FuncGet(ctx, scd, scd.ChunkMetas[idx].URL, headers, scd.sc.rest.RequestTimeout)
 	if err != nil {
-		raiseDownloadError(scd, idx, err)
-		return
+		return err
 	}
 	bufStream := bufio.NewReader(resp.Body)
 	defer resp.Body.Close()
 	glog.V(2).Infof("response returned chunk: %v, resp: %v", idx+1, resp)
-	if resp.StatusCode == http.StatusOK {
-		gzipMagic, err := bufStream.Peek(2)
+	if resp.StatusCode != http.StatusOK {
+		b, err := ioutil.ReadAll(bufStream)
 		if err != nil {
-			raiseDownloadError(scd, idx, err)
-			return
+			return err
 		}
-		start := time.Now()
-		var source io.Reader
-		if gzipMagic[0] == 0x1f && gzipMagic[1] == 0x8b {
-			// detects and uncompresses Gzip format data
-			bufStream0, err := gzip.NewReader(bufStream)
-			if err != nil {
-				raiseDownloadError(scd, idx, err)
-				return
-			}
-			defer bufStream0.Close()
-			source = bufStream0
-		} else {
-			source = bufStream
+		glog.V(1).Infof("HTTP: %v, URL: %v, Body: %v", resp.StatusCode, scd.ChunkMetas[idx].URL, b)
+		glog.V(1).Infof("Header: %v", resp.Header)
+		glog.Flush()
+		return &SnowflakeError{
+			Number:      ErrFailedToGetChunk,
+			SQLState:    SQLStateConnectionFailure,
+			Message:     errMsgFailedToGetChunk,
+			MessageArgs: []interface{}{idx},
 		}
-		st := &largeResultSetReader{
-			status: 0,
-			body:   source,
```

Comment:
@sfc-gh-jbahk I recall this is required for Azure Blob support. Can you check the history?
https://github.com/snowflakedb/gosnowflake/commit/2264cec1

Reply:
I moved the core functionality into a separate method because the method was getting too long. The `decodeChunk` method takes care of decoding (whether it be JSON, Arrow or the custom JSON decoder). The functionality should be the same.

Reply:
hmm, it is interesting... The GH action is passing for GCP/GCS. @sfc-gh-jbahk , can you check if this is no longer required? The problem was since GCS didn't set the header properly, the clients could not get the right content type.

Reply:
@sfc-gh-jbahk , if you don't have access to GCS account, let me know. You could use the local dev gcsaccount.

Reply:
ah, ok, so it is checking the Gzip header. I took wrong.